.github/workflows/ci-build.yml

```diff
@@ -315,3 +315,41 @@ jobs:
       run: |
         cd ${GITHUB_WORKSPACE}/build
         ctest --timeout 120
+
+  check_annotations:
+    name: Check Annotations
+    needs:
+    - emscripten
+    - linux
+    - macos
+    - windows
+
+    if: github.event_name == 'pull_request'
+
+    runs-on: ubuntu-20.04
+
+    steps:
+    - name: Get check suite ID
+      id: check_suite_id
+      uses: octokit/request-action@v2.x
+      with:
+        route: GET /repos/{repository}/actions/runs/{run_id}
+        repository: ${{ github.repository }}
+        run_id: ${{ github.run_id }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Get check runs
+      id: check_runs
+      uses: octokit/request-action@v2.x
+      with:
+        route: GET /repos/{repository}/check-suites/{check_suite_id}/check-runs
+        repository: ${{ github.repository }}
+        check_suite_id: ${{ fromJson(steps.check_suite_id.outputs.data).check_suite_id }}
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Check annotations
+      shell: bash
+      run: |
+        exit $(echo '${{ toJson(fromJson(steps.check_runs.outputs.data).check_runs.*.output.annotations_count) }}' | jq 'reduce .[] as $i (0; . + $i)')
```

Comment:
Given we have the JSON output here .. can't we do something fun with it, like reporting how many annotations there were? Simple embedded Python script of sorts?

PS: I really like what you did here, this is just bikeshedding :)